### DISCUSSION TITLE
Initial import of music library containing both albums and singles

### DISCUSSION THREAD

**Discussion by SamuelCook** [beetbox/beets#4136] [Q&A]
My music library is a collection of both albums and singles. Singles reside within an 'Unknown Album' directory within each artist.

What is the recommended approach to load this collection into beets?

It would be nice if the import path would support wildcards, but it doesn't appear to:
```shell
$ beet import -ips '/music/*/Unknown Album/*'
error: no such file or directory: /music/*/Unknown Album/*
```

I've also tried importing without singleton mode enabled, and selecting 'as Tracks' for each directory of singles. However, beets does not appear to remember this choice, and asks me on each subsequent import how to handle the same files. 

```shell
$> beets beet import -ip /music

music/1200 Techniques/Unknown Album (1 items)
Finding tags for album "1200 Techniques - ".
Candidates:
1. Various Artists - Radioactivity Volume 02-05 (13.5%) (missing tracks, album) (CD, 2002, AU, SBA Music, RAD0205)
2. Various Artists - So Fresh: The Hits of Winter 2002 (13.0%) (missing tracks, album) (CD, 2002, AU, Columbia, 5083842000)
3. Various Artists - The Essential Noughties 2000 - 2005 (8.5%) (missing tracks, album, year) (2xCD, 2006, AU, Sony BMG Music Entertainment, 82876850992)
4. Various Artists - 20 Years of Rage (3.6%) (missing tracks, album, year) (4xCD, 2007, AU, ABC Music, 1778704)
5. Various Artists - 20 Years of Rage (3.6%) (missing tracks, album, year) (4xCD, 2007, AU, Warner Music Australia, 5144211572)
# selection (default 1), Skip, Use as-is, as Tracks, Group albums,
Enter search, enter Id, aBort? T

/music/1200 Techniques/Unknown Album/Karma.mp3
Tagging track: 1200 Techniques - Karma
URL:
    https://musicbrainz.org/recording/1cd193c2-67ad-4143-beaa-69a0afff0f06
(Similarity: 100.0%)

/music/28 Days/Unknown Album (4 items)
Finding tags for album "28 Days - ".
Candidates:
1. 28 Days - 10 Years of Cheap Fame (44.6%) (missing tracks, album, year) (CD, 2007, AU, DD0010)
2. 28 Days - Stealing Chairs (24.5%) (missing tracks, album, tracks, ...) (2002)
3. 28 Days - Stealing Chairs (24.2%) (missing tracks, tracks, album, ...) (2002, AU)
# selection (default 1), Skip, Use as-is, as Tracks, Group albums,
Enter search, enter Id, aBort? B

$> beets beet import -ip /music
Resuming interrupted import of /music

/music/1200 Techniques/Unknown Album (1 items)
Finding tags for album "1200 Techniques - ".
Candidates:
1. Various Artists - Radioactivity Volume 02-05 (13.5%) (missing tracks, album) (CD, 2002, AU, SBA Music, RAD0205)
2. Various Artists - So Fresh: The Hits of Winter 2002 (13.0%) (missing tracks, album) (CD, 2002, AU, Columbia, 5083842000)
3. Various Artists - The Essential Noughties 2000 - 2005 (8.5%) (missing tracks, album, year) (2xCD, 2006, AU, Sony BMG Music Entertainment, 82876850992)
4. Various Artists - 20 Years of Rage (3.6%) (missing tracks, album, year) (4xCD, 2007, AU, ABC Music, 1778704)
5. Various Artists - 20 Years of Rage (3.6%) (missing tracks, album, year) (4xCD, 2007, AU, Warner Music Australia, 5144211572)
# selection (default 1), Skip, Use as-is, as Tracks, Group albums,
Enter search, enter Id, aBort?
```

**Comment by sampsyo** [comment]
The importer doesn’t support wildcards, but your shell probably does. If you remove those single quotes, you can pass multiple directories into the import command line.

**Comment by SamuelCook** [comment]
That's a really good point...

I've been using beets from within a docker container, so of course the wildcards didn't work when I executed it from the host via `docker exec`... because on the host, /music does not exist. I have now run `docker exec beets bash` and started the import in singleton mode without quotes, and it seems to be working great.

However, it did skip the files I'd already imported as tracks. To test a theory, I imported a bunch of singletons files, then aborted the import and ran a full import in album mode again, and sure enough, beets asked me what to do with the singleton tracks that I'd already imported:

```
$> beet import -ip /music
Resuming interrupted import of /music

/music/1200 Techniques/Unknown Album (1 items)
Finding tags for album "1200 Techniques - ".
Candidates:
1. Various Artists - Radioactivity Volume 02-05 (13.5%) (missing tracks, album) (CD, 2002, AU, SBA Music, RAD0205)
2. Various Artists - So Fresh: The Hits of Winter 2002 (13.0%) (missing tracks, album) (CD, 2002, AU, Columbia, 5083842000)
3. Various Artists - The Essential Noughties 2000 - 2005 (8.5%) (missing tracks, album, year) (2xCD, 2006, AU, Sony BMG Music Entertainment, 82876850992)
4. Various Artists - 20 Years of Rage (3.6%) (missing tracks, album, year) (4xCD, 2007, AU, ABC Music, 1778704)
5. Various Artists - 20 Years of Rage (3.6%) (missing tracks, album, year) (4xCD, 2007, AU, Warner Music Australia, 5144211572)
# selection (default 1), Skip, Use as-is, as Tracks, Group albums,
Enter search, enter Id, aBort? B
```

Will I have to perform further command line trickery to import while excluding the `Unknown Album` dirs? Is there a simpler approach? Shouldn't beets just ignore files it's already imported?

**Reply by sampsyo** [reply]
The key here is this message: `Resuming interrupted import of /music`

The importer tries to "fast forward" to one past the most recently imported directory in an attempt to help recover from crashes (and aborts). If you want to avoid importing stuff that you previously imported in other runs, the `incremental` mode for the importer might be want you want—it instead marks directories as imported and avoids importing them again. To be honest, however, I'm not sure how well that mode works with singletons.

One option might be to try setting the `ignore` configuration option for the second run to skip directories named "Unknown Album".
